src/qt/managenamespage.cpp

```diff
@@ -98,6 +115,54 @@ void ManageNamesPage::onCopyValueAction()
     GUIUtil::copyEntryData(ui->tableView, NameTableModel::Value);
 }
 
+void ManageNamesPage::onRenewNameAction()
+{
+    QModelIndexList indexes = GUIUtil::getEntryData(ui->tableView, NameTableModel::Name);
+
+    if (indexes.isEmpty())
+        return;
+
+    QString msg;
+    QString title;
+
+    if (indexes.size() == 1)
+    {
+        const QString &name = indexes.at(0).data(Qt::EditRole).toString();
+
+        msg = tr ("Are you sure you want to renew the name <b>%1</b>?")
+            .arg (GUIUtil::HtmlEscape (name));
+        title = tr ("Confirm name renewal");
+    }
+    else
+    {
+        msg = tr ("Are you sure you want to renew multiple names simultaneously?  This will reveal common ownership of the renewed names (bad for anonymity).");
+        title = tr ("Confirm multiple name renewal");
+    }
+
+    QMessageBox::StandardButton res;
+    res = QMessageBox::question (this, title, msg,
+                                 QMessageBox::Yes | QMessageBox::Cancel,
+                                 QMessageBox::Cancel);
+    if (res != QMessageBox::Yes)
+        return;
+
+    WalletModel::UnlockContext ctx(walletModel->requestUnlock ());
+    if (!ctx.isValid ())
+        return;
+
+    for (QModelIndexList::iterator i = indexes.begin(); i != indexes.end(); i++)
```

Comment:
Out of curiosity:  Is it possible to use a C++11 "for each" loop here (I don't konw the details of `QModelIndexList`)?

Reply:
Good catch.  I just ran `grep -r QModelIndex ./src/qt | grep for`, and the existing three cases from upstream all use a for each loop.  I'll adjust this code to match upstream style.